src/base59.cpp

```diff
@@ -0,0 +1,150 @@
+#include <assert.h>
+#include <limits>
+#include <stdexcept>
+#include <string.h>
+#include <vector>
+#include <veriblock/base58.hpp>
+#include <string>
+
+namespace VeriBlock {
+
+static std::string g_Base59Alphabet =
+    "123456789ABCDEFGHJKLMNPQRSTUVWXYZabcdefghijkmnopqrstuvwxyz0";
+static std::vector<uint32_t> g_Indexes(128);
+static const uint32_t g_kBase_256 = 256;
+static const size_t g_kBase59 = g_Base59Alphabet.size();
+
+static uint8_t divmod59(std::vector<uint8_t> &number, size_t startAt) {
+    size_t remainder = 0;
+  for (size_t i = startAt; i < number.size(); i++) {
+    uint32_t digit256 = (uint32_t)number[i] & 0xFF;
+    size_t temp = remainder * g_kBase_256 + digit256;
+    number[i] = (uint8_t)(temp / g_kBase59);
+    remainder = temp % g_kBase59;
+  }
+  return (uint8_t)remainder;
+}
+
+static uint8_t divmod256(std::vector<uint8_t> &number59, size_t startAt) {
+    size_t remainder = 0;
+  for (size_t i = startAt; i < number59.size(); i++) {
+    int digit59 = (int8_t)number59[i] & 0xFF;
+    size_t temp = remainder * g_kBase59 + digit59;
+    number59[i] = (uint8_t)(temp / g_kBase_256);
+    remainder = temp % g_kBase_256;
+  }
+  return (uint8_t)remainder;
+}
+
+std::string EncodeBase59(const void *buf, size_t nSize) {
```

Comment:
```suggestion
std::string EncodeBase59(const uint8_t *buf, size_t nSize) {
```
anyway, we use this as byte array

Reply:
fixed